src/Explorer/Controls/Settings/SettingsComponent.tsx

```diff
@@ -0,0 +1,2669 @@
+              value: originalThroughputValue,
+              updateCurrent: true,
+              updateBaseline: true
+            });
+
+            this.setState({
+              notificationStatusInfo: this.throughputApplyDelayedMessage(
+                this.state.isAutoPilotSelected,
+                originalThroughputValue,
+                this.getThroughputUnit(),
+                this.collection.databaseId,
+                this.collection.id(),
+                newThroughput
+              )
+            });
+          } catch (error) {
+            TelemetryProcessor.traceFailure(
+              Action.UpdateSettings,
+              {
+                databaseAccountName: this.container.databaseAccount().name,
+                databaseName: this.collection && this.collection.databaseId,
+                collectionName: this.collection && this.collection.id(),
+                defaultExperience: this.container.defaultExperience(),
+                dataExplorerArea: Constants.Areas.Tab,
+                tabTitle: this.props.settingsTab.tabTitle(),
+                error: error
+              },
+              startKey
+            );
+            throw error;
+          }
+        } else {
+          const updatedOffer: DataModels.Offer = await updateOffer(this.collection.offer(), newOffer, headerOptions);
+          this.collection.offer(updatedOffer);
+          this.collection.offer.valueHasMutated();
+        }
+      }
+      this.container.isRefreshingExplorer(false);
+      this.setBaseline();
+      this.collection.readSettings();
+      this.setState({ wasAutopilotOriginallySet: this.state.isAutoPilotSelected });
+      TelemetryProcessor.traceSuccess(
+        Action.UpdateSettings,
+        {
+          databaseAccountName: this.container.databaseAccount().name,
+          defaultExperience: this.container.defaultExperience(),
+          dataExplorerArea: Constants.Areas.Tab,
+          tabTitle: this.props.settingsTab.tabTitle()
+        },
+        startKey
+      );
+    } catch (reason) {
+      this.container.isRefreshingExplorer(false);
+      this.props.settingsTab.isExecutionError(true);
+      console.error(reason);
+      TelemetryProcessor.traceFailure(
+        Action.UpdateSettings,
+        {
+          databaseAccountName: this.container.databaseAccount().name,
+          defaultExperience: this.container.defaultExperience(),
+          dataExplorerArea: Constants.Areas.Tab,
+          tabTitle: this.props.settingsTab.tabTitle()
+        },
+        startKey
+      );
+    }
+    this.props.settingsTab.isExecuting(false);
+  };
+
+  private onRevertClick = (): void => {
+    this.updateStatefulValue({
+      key: StatefulValueNames.Throughput,
+      value: this.state.throughput.baseline,
+      updateCurrent: true
+    });
+    this.updateStatefulValue({
+      key: StatefulValueNames.TimeToLive,
+      value: this.state.timeToLive.baseline,
+      updateCurrent: true
+    });
+    this.updateStatefulValue({
```

Comment:
so StatefulValue is a new interface I've defined modelled around ViewModels.Editable, used in ko version of the SettingsTab. These StatefulValue<> objects are part of the state of the SettingsTab, and are modified when controls are changed in the SettingsTab.
For example, for modifying throughput:
1) SettingsComponentState has a "throughput" field of type StatefulValue<number>
2) baseline and current values for state.throughput are set to 'undefined' in the constructor
3) when saving, this.state.throughput.current is used
4) when resetting, this.state.throughput.current is set to this.state.throughput.baseline (in the updateStatefulValue function)
5) When modifying, this.state.thoughput.current is set to the new value from the input (using updateStatefulValue)

updateStatefulValue function was introduced to reuse code since the same functionality is replicated for the different state fields that need both baseline and current values tracked.